client/network/src/chain.rs

```diff
@@ -19,18 +19,25 @@
 //! Blockchain access trait
 
 use sp_blockchain::{Error, HeaderBackend, HeaderMetadata};
-use sc_client_api::{BlockBackend, ProofProvider};
+use sc_client_api::{
+	backend::{Backend as BackendT, StorageProvider},
+	BlockBackend, ProofProvider
+};
 use sp_runtime::traits::{Block as BlockT, BlockIdTo};
 
 /// Local client abstraction for the network.
-pub trait Client<Block: BlockT>: HeaderBackend<Block> + ProofProvider<Block> + BlockIdTo<Block, Error = Error>
-	+ BlockBackend<Block> + HeaderMetadata<Block, Error = Error> + Send + Sync
+pub trait Client<Block: BlockT, Backend>: HeaderBackend<Block> + ProofProvider<Block> + BlockIdTo<Block, Error = Error>
+	+ BlockBackend<Block> + HeaderMetadata<Block, Error = Error> + StorageProvider<Block, Backend> + Send + Sync
+	where
+		Backend: BackendT<Block>
```

Comment:
Assuming that we leave the storage queries in `sc-service` (which I don't strongly prefer), please revert the changes to this trait.